waspc/src/Wasp/Generator/WebAppGenerator.hs

```diff
@@ -142,48 +149,59 @@ genPublicIndexHtml spec =
-srcDir = C.webAppSrcDirInWebAppRootDir
-
 -- TODO(matija): Currently we also generate auth-specific parts in this file (e.g. token management),
 -- although they are not used anywhere outside.
 -- We could further "templatize" this file so only what is needed is generated.
 --
 
--- | Generates api.js file which contains token management and configured api (e.g. axios) instance.
-genApi :: FileDraft
-genApi = C.mkTmplFd (C.asTmplFile [relfile|src/api.js|])
-
 genSrcDir :: AppSpec -> Generator [FileDraft]
-genSrcDir spec = do
-  routerFd <- RouterGenerator.generateRouter spec
-  operationsFds <- genOperations spec
-  authFds <- AuthG.genAuth spec
+genSrcDir spec =
+  sequence [genRouter, genIndexJs spec, genApi]
+    <++> genDirectCopies
+    <++> genOperationsDir
+    <++> genAuthDir
+  where
+    genRouter = RouterGenerator.generateRouter spec
+    genOperationsDir = OperationsG.genOperations spec
+    genAuthDir = AuthG.genAuth spec
 
+genDirectCopies :: Generator [FileDraft]
+genDirectCopies =
   return $
-    generateLogo :
-    routerFd :
-    genApi :
     map
-      processSrcTmpl
-      [ [relfile|index.js|],
-        [relfile|index.css|],
+      mkSrcTmplFd
+      [ [relfile|index.css|],
+        [relfile|logo.png|],
         [relfile|serviceWorker.js|],
         [relfile|config.js|],
         [relfile|queryClient.js|],
         [relfile|utils.js|]
       ]
-      ++ operationsFds
-      ++ authFds
+
+-- | Generates api.js file which contains token management and configured api (e.g. axios) instance.
+genApi :: Generator FileDraft
+genApi = return $ C.mkTmplFd (C.asTmplFile [relfile|src/api.js|])
+
+-- | TODO(filip): we have almost the same thing in server
+genIndexJs :: AppSpec -> Generator FileDraft
+genIndexJs spec =
+  return $
+    C.mkTmplFdWithDstAndData
+      (asTmplFile [relfile|src/index.js|])
+      (asWebAppFile [relfile|src/index.js|])
+      ( Just $
+          object
+            [ "doesClientSetupFnExist" .= isJust maybeSetupJsFunction,
+              "clientSetupJsFnImportStatement" .= fromMaybe "" maybeSetupJsFnImportStmt,
+              "clientSetupJsFnIdentifier" .= fromMaybe "" maybeSetupJsFnImportIdentifier
+            ]
+      )
   where
-    generateLogo =
-      C.mkTmplFdWithDstAndData
-        (asTmplFile [relfile|src/logo.png|])
-        (srcDir </> asWebAppSrcFile [relfile|logo.png|])
-        Nothing
-    processSrcTmpl path =
-      C.mkTmplFdWithDst
-        (asTmplFile $ [reldir|src|] </> path)
-        (srcDir </> asWebAppSrcFile path)
+    maybeSetupJsFunction = AS.App.Client.setupFn =<< AS.App.client (snd $ getApp spec)
+    maybeSetupJsFnImportDetails = getJsImportDetailsForExtFnImport relPosixPathFromSrcDirToExtSrcDir <$> maybeSetupJsFunction
+    (maybeSetupJsFnImportIdentifier, maybeSetupJsFnImportStmt) =
+      (fst <$> maybeSetupJsFnImportDetails, snd <$> maybeSetupJsFnImportDetails)
+
+-- TODO(filip): Move this somewhere common (we have the same thing in server)
```

Comment:
Although keep in mind these are different things -> one is path to external code dir in web client, another is path to external code dir in server. They are the same path, true, but still they could be different values in theory.

Reply:
Right, that's true. I removed the todo for now.

Reply:
My mistake, didn't commit that one yet.